ipld/resolver/src/observe.rs

```diff
@@ -0,0 +1,331 @@
+
+    IPLD_RESOLVER_DISCOVERY_BACKGROUND_LOOKUP: IntCounter =
+        register_int_counter!("ipld_resolver_discovery_background_lookup", "Number of background lookups started");
+
+    IPLD_RESOLVER_DISCOVERY_CONNECTED_PEERS: IntGauge =
+        register_int_gauge!("ipld_resolver_discovery_connected_peers", "Number of connections");
+
+    IPLD_RESOLVER_MEMBERSHIP_SKIPPED_PEERS: IntCounter =
+        register_int_counter!("ipld_resolver_membership_skipped_peers", "Number of providers skipped");
+
+    IPLD_RESOLVER_MEMBERSHIP_ROUTABLE_PEERS: IntGauge =
+        register_int_gauge!("ipld_resolver_membership_routable_peers", "Number of routable peers");
+
+    IPLD_RESOLVER_MEMBERSHIP_PROVIDER_PEERS: IntGauge =
+        register_int_gauge!("ipld_resolver_membership_provider_peers", "Number of unique providers");
+
+    IPLD_RESOLVER_MEMBERSHIP_UNKNOWN_TOPIC: IntCounter =
+        register_int_counter!("ipld_resolver_membership_unknown_topic", "Number of messages with unknown topic");
+
+    IPLD_RESOLVER_MEMBERSHIP_INVALID_MESSAGE: IntCounter =
+        register_int_counter!("ipld_resolver_membership_invalid_message", "Number of invalid messages received");
+
+    IPLD_RESOLVER_MEMBERSHIP_PUBLISH_SUCCESS: IntCounter =
+        register_int_counter!("ipld_resolver_membership_publish_total", "Number of published messages");
+
+    IPLD_RESOLVER_MEMBERSHIP_PUBLISH_FAILURE: IntCounter =
+        register_int_counter!("ipld_resolver_membership_publish_failure", "Number of failed publish attempts");
+
+    IPLD_RESOLVER_CONTENT_RESOLVE_RUNNING: IntGauge =
+        register_int_gauge!("ipld_resolver_content_resolve_running", "Number of currently running content resolutions");
+
+    IPLD_RESOLVER_CONTENT_RESOLVE_NO_PEERS: IntCounter =
+        register_int_counter!("ipld_resolver_content_resolve_no_peers", "Number of resolutions with no known peers");
+
+    IPLD_RESOLVER_CONTENT_RESOLVE_SUCCESS: IntCounter =
+        register_int_counter!("ipld_resolver_content_resolve_success", "Number of successful resolutions");
+
+    IPLD_RESOLVER_CONTENT_RESOLVE_FAILURE: IntCounter =
+        register_int_counter!("ipld_resolver_content_resolve_failure", "Number of failed resolutions");
+
+    IPLD_RESOLVER_CONTENT_RESOLVE_FALLBACK: IntCounter =
+        register_int_counter!("ipld_resolver_content_resolve_fallback", "Number of resolutions that fall back on secondary peers");
+
+    IPLD_RESOLVER_CONTENT_RESOLVE_PEERS: Histogram =
+        register_histogram!("ipld_resolver_content_resolve_peers", "Number of peers found for resolution from a subnet");
+
+    IPLD_RESOLVER_CONTENT_CONNECTED_PEERS: Histogram =
+        register_histogram!("ipld_resolver_content_connected_peers", "Number of connected peers in a resolution");
+
+    IPLD_RESOLVER_CONTENT_RATE_LIMITED: IntCounter =
+        register_int_counter!("ipld_resolver_content_rate_limited", "Number of rate limited requests");
+}
+
+impl_traceables!(TraceLevel::Info, "IPLD_Resolver/Ping", PingEvent);
```

Comment:
We actually allow users to filter by the domain. I think having separate domain for each event type makes it impossible to filter for all events from resolver.

Reply:
So are you suggesting that the domain will be simply `IPLD_Resolver`? because previously (before my last commit) they were also segregated (e.g., `Ping`, `Discovery`, etc.).
Another option is to implement domain namespace filtering, but it might be an overkill.

Reply:
Yeah, I think having a unified domain might work better. We do allow to filter by it.

Reply:
> Another option is to implement domain namespace filtering, but it might be an overkill.

Yes. I think it would be.

Reply:
fixed here: https://github.com/consensus-shipyard/ipc/pull/1207/commits/8b5051c286a09ec61d3f2c20ffcd0446f87344a1